File: DDD DSL/DDD DSL Design War FizzBuzzWhizz Reloaded/Code/Java/src/fayelab/ddd/fbwreloaded/interpreter/Predication.java
package fayelab.ddd.fbwreloaded.interpreter;

public class Predication
{
    enum Type {TIMES, CONTAINS, ALWAYSTRUE};
    
    private Type type;
    private int param;
    
    public Predication(Type type, int param)
    {
        this.type = type;
        this.param = param;
    }

    public Type getType()
    {
        return type;
    }

    public int getParam()
    {
        return param;
    }
}

File: DDD DSL/DDD DSL Design War FizzBuzzWhizz/Code/Python/eight/straightforward/predication.py
class Predication(object):
    def predicate(self, n):
        pass

class Times(Predication):
    def __init__(self, base):
        self.base = base

    def predicate(self, n):
        return n % self.base == 0

class Contains(Predication):
    def __init__(self, digit):
        self.digit = digit

    def predicate(self, n):
        return str(self.digit) in str(n)

class AlwaysTrue(Predication):
    def predicate(self, n):
        return True